cmd/cost/list.go

```diff
@@ -142,6 +182,87 @@ func listCostsUnderOU(OU *organizations.OrganizationalUnit, awsClient awsprovide
 	return nil
 }
 
+type AccountCost struct {
+	AccountID string
+	Cost      float64
+	Unit      string
+}
+
+type OUCost struct {
+	Costs   []AccountCost
+	OU      *organizations.OrganizationalUnit
+	options *listOptions
+}
+
+func (o *OUCost) getCost(awsClient awsprovider.Client) error {
+
+	accounts, err := getAccountsRecursive(o.OU, awsClient)
+	if err != nil {
+		return err
+	}
+
+	ops := &getOptions{
+		time:  o.options.time,
+		start: o.options.start,
+		end:   o.options.end,
+		ou:    o.options.ou,
+	}
+
+	for _, account := range accounts {
+		accCost := AccountCost{
+			AccountID: *account,
+			Unit:      "",
+			Cost:      0,
+		}
+		err = ops.getAccountCost(account, &accCost.Unit, awsClient, &accCost.Cost)
+		if err != nil {
+			return err
+		}
+		o.Costs = append(o.Costs, accCost)
+	}
+
+	sort.Slice(o.Costs, func(i, j int) bool {
+		return o.Costs[i].Cost < o.Costs[j].Cost
+	})
+
+	return nil
+}
+
+func (o OUCost) getSum() (sum float64, unit string, err error) {
+	sum = 0
+	for _, cost := range o.Costs {
+		sum += cost.Cost
+		if unit == "" {
+			unit = cost.Unit
+			continue
+		}
+		if unit != cost.Unit {
+			err = fmt.Errorf("can't sum up different currencies: %s and %s", unit, cost.Unit)
+			return
+		}
+	}
+	return
+}
+
+func (o OUCost) printCostPerAccount() {
+
+	if o.options.csv {
+		fmt.Println("AccountID,Cost,Unit")
+	}
+	for _, accountCost := range o.Costs {
+		resp := listAccountCostResponse{
+			AccountId: accountCost.AccountID,
+			Cost:      accountCost.Cost,
+			Unit:      accountCost.Unit,
+		}
+		if o.options.csv {
+			fmt.Printf("%s,%.2f,(%s)\n", accountCost.AccountID, accountCost.Cost, accountCost.Unit)
```

Comment:
Shouldn't this be using the o.options value?

Reply:
which value?

Reply:
The output file
As printf prints to stdout and not to the provided output device

Reply:
makes sense I think

Reply:
Ok